src/string_pool.cr

```diff
@@ -1,19 +1,52 @@
+# A String pool is a collection of Strings
+# It can be used to return an already present String instead of creating a new String.
+#
+#
+#   require "string_pool"
+#   stringpool = StringPool.new
+
 class StringPool
+  # Returns the size
+  #
+  #     stringpool = StringPool.new
+  #     puts stringpool.size # => 0
   getter size
 
+  # Initialization of the elements 
+  # At the begining, the size is null and the bucket is an array of size 11
   def initialize
     @buckets = Array(Array(String)?).new(11, nil)
     @size = 0
   end
 
+  # Returns true if the String Pool have no element else returns false
```

Comment:
"have no element" -> "has no elements"

And maybe add a comma before "else" and use "otherwise" instead.